### ISSUE TITLE
[feature] (File History) diff against any base rev

### ISSUE THREAD

**Issue by eshepelyuk** [sindrets/diffview.nvim#95]
It would be great to extend file history view ( invoked by `:DiffviewFileHistory <FILE_NAME>`) with possibility to show diff not only before and after particular commit, but also to have difference between particular commit and current (`HEAD`) version of a file.

The current version should be editable, so one could navigate through history of commits and be able to pick up particular changes into current version.

Maybe it can be additional option to file history view, that switched between modes

* when `true` - current diff behaviour, showing differences caused by a single commit.
* when `false` - then show differene from a commit to current version, with editable buffer of current file version.

**Comment by sindrets** [comment]
It sounds like this use-case is already covered by features that already exist. See:

- `:h diffview-maps-goto_file`
- `:h diffview-maps-goto_file_split`
- `:h diffview-maps-goto_file_tab`

Using these mappings you can quickly bring up the local version of a file.

Another way to solve this would be to use the `y` mapping from the file history panel to copy the commit hash for the entry under the cursor. Then you can open a full diffview for that commit with `:DiffviewOpen <commit-SHA>`.

**Comment by eshepelyuk** [comment]
`y` and then  `:DiffviewOpen <commit-SHA>` sort of solving the task, but quite infomcortable to switch if one need to pick up dozens of changes from previous version of a single file.

the `diffview-maps-goto*` are not suitable since they open a new buffer, while what I asked is to load a diff view of two files

**Comment by eshepelyuk** [comment]
What I am asking for is a shortcut from file history file list to execute smth like

`:DiffviewOpen <commit-SHA of current entry> -- <name of current entry>`, but without actual swtich to `DiffviewOpen` mode.

**Comment by sindrets** [comment]
Okay, the only way I see this being properly useful is if we can specify any base rev to diff against. Something like:
`:DiffviewFileHistory --base=<REV> <PATHS>`

**Comment by eshepelyuk** [comment]
Seems ok to me, thnx.

**Comment by sindrets** [comment]
@eshepelyuk This has now been implemented through #103. For your use-case you can call: 

```
:DiffviewFileHistory --base=LOCAL
``` 

This will bring up the file history with the right side of the diff split always being the local state of the file. If you want this to be the default behavior you can just add that arg to your `default_args` in the config, like:

```lua
require("diffview").setup({
  default_args = {
    DiffviewFileHistory = { "--base=LOCAL" },
  },
})
```

**Comment by ievgenii-shepeliuk** [comment]
Thnx @sindrets, works as expected.
